### ISSUE TITLE
Connection request dialog box does not appear (not Asus)

### ISSUE THREAD

**Issue by BLABLA2003** [Genymobile/gnirehtet#161]
I have seen issues with this problem but they were all on Asus phones and i aslo noticed that something something is different in the start.
I would sent a adb logcat but there is a continues check of state and can`t make out anything relevant.
This is the message i get on start.
```
2018-12-11 20:35:07.694 INFO Main: Starting relay server...
2018-12-11 20:35:07.699 INFO Relay: Relay server started
2018-12-11 20:35:07.700 INFO Main: Checking gnirehtet client...
2018-12-11 20:35:07.823 INFO Main: Starting client...
Broadcasting: Intent { act=com.genymobile.gnirehtet.START flg=0x400000 cmp=com.genymobile.gnirehtet/.GnirehtetControlReceiver }
Broadcast completed: result=0
```

I had reinstalled it and tried again the same thing happens.
Phone specs:
Nokia 8
android Oreo 8.1.0

**Comment by rom1v** [comment]
This problem occurs too often on several devices.

In the end, I think I will change to send all commands to an `Activity` instead of a `BroadcastReceiver`.

**Comment by rom1v** [comment]
> adb logcat but there is a continues check of state and can`t make out anything relevant

What about `adb logcat | grep gnirehtet` while you start gnirehtet?

**Comment by BLABLA2003** [comment]
it says that grep is not a command

**Comment by rom1v** [comment]
@BLABLA2003 Oh, this is because you are not on Linux.

Make it execute on the device instead:

    adb shell 'logcat | grep gnirehtet'

**Comment by BLABLA2003** [comment]
the same thing

**Comment by rom1v** [comment]
OK, do the way you want, but keep (or look for) the lines containing `gnirehtet` in the output of `adb logcat` :wink:

**Comment by disadone** [comment]
I have the same problem . I em... got this.

> 12-14 19:49:23.902  1160  2858 I ActivityManager: START u0 {act=android.intent.action.VIEW dat=file:///data/local/tmp/gnirehtet.apk typ=application/vnd.android.package-archive flg=0x18000000 cmp=com.miui.securitycenter/com.miui.permcenter.install.AdbInstallActivity (has extras)} from uid 2000 from pid 6008 on display 0
12-14 19:49:27.645  6087  6097 D DefContainer: Copying /data/local/tmp/gnirehtet.apk to base.apk
12-14 19:49:28.535  1160  2858 W BroadcastQueueInjector: Unable to launch app com.android.providers.downloads.ui/10009 for broadcast Intent { act=android.intent.action.PACKAGE_ADDED dat=package:com.genymobile.gnirehtet flg=0x4000010 (has extras) }: process is not permitted to  auto start
12-14 19:49:28.861  1160  1228 W XSpaceManagerService: update XSpace App: packageName:com.genymobile.gnirehtet, user:UserHandle{0}, action:android.intent.action.PACKAGE_ADDED
12-14 19:49:29.183  1160  1181 W BroadcastQueueInjector: Unable to launch app com.xiaomi.gamecenter/10052 for broadcast Intent { act=android.intent.action.PACKAGE_ADDED dat=package:com.genymobile.gnirehtet flg=0x4000010 (has extras) }: process is not permitted to  auto start
12-14 19:49:29.186  1160  1222 W BroadcastQueueInjector: Unable to launch app com.xiaomi.gamecenter/10052 for broadcast Intent { act=android.intent.action.PACKAGE_ADDED dat=package:com.genymobile.gnirehtet flg=0x4000010 (has extras) }: process is not permitted to  auto start
12-14 19:49:29.187  1160  1222 W BroadcastQueueInjector: Unable to launch app com.wm.dmall/10109 for broadcast Intent { act=android.intent.action.PACKAGE_ADDED dat=package:com.genymobile.gnirehtet flg=0x4000010 (has extras) }: process is not permitted to  auto start
12-14 19:49:29.188  1160  1222 W BroadcastQueueInjector: Unable to launch app com.ifeng.news2/10134 for broadcast Intent { act=android.intent.action.PACKAGE_ADDED dat=package:com.genymobile.gnirehtet flg=0x4000010 (has extras) }: process is not permitted to  auto start
12-14 19:49:29.189  1160  1222 W BroadcastQueueInjector: Unable to launch app cn.xichan.youquan.ui/10049 for broadcast Intent { act=android.intent.action.PACKAGE_ADDED dat=package:com.genymobile.gnirehtet flg=0x4000010 (has extras) }: process is not permitted to  auto start
12-14 19:49:29.554  1160  1222 W BroadcastQueue: Permission Denial: broadcasting Intent { act=com.genymobile.gnirehtet.START flg=0x10 cmp=com.genymobile.gnirehtet/.GnirehtetControlReceiver } from null (pid=6196, uid=2000) requires android.permission.WRITE_SECURE_SETTINGS due to receiver com.genymobile.gnirehtet/.GnirehtetControlReceiver
12-14 19:50:20.154  1160  1222 W BroadcastQueue: Permission Denial: broadcasting Intent { act=com.genymobile.gnirehtet.STOP flg=0x10 cmp=com.genymobile.gnirehtet/.GnirehtetControlReceiver } from null (pid=6314, uid=2000) requires android.permission.WRITE_SECURE_SETTINGS due to receiver com.genymobile.gnirehtet/.GnirehtetControlReceiver
12-14 19:50:22.542  1160  1222 W BroadcastQueue: Permission Denial: broadcasting Intent { act=com.genymobile.gnirehtet.START flg=0x10 cmp=com.genymobile.gnirehtet/.GnirehtetControlReceiver } from null (pid=6326, uid=2000) requires android.permission.WRITE_SECURE_SETTINGS due to receiver com.genymobile.gnirehtet/.GnirehtetControlReceiver
12-14 19:54:44.257  6561  6561 I ADB_SERVICES: service_to_fd shell:am broadcast -a com.genymobile.gnirehtet.STOP -n com.genymobile.gnirehtet/.GnirehtetControlReceiver
12-14 19:54:44.900  1160  1222 W BroadcastQueue: Permission Denial: broadcasting Intent { act=com.genymobile.gnirehtet.STOP flg=0x10 cmp=com.genymobile.gnirehtet/.GnirehtetControlReceiver } from null (pid=7197, uid=2000) requires android.permission.WRITE_SECURE_SETTINGS due to receiver com.genymobile.gnirehtet/.GnirehtetControlReceiver
12-14 19:54:55.674  6561  6561 I ADB_SERVICES: service_to_fd shell:dumpsys package com.genymobile.gnirehtet
12-14 19:54:55.750  6561  6561 I ADB_SERVICES: service_to_fd reverse:forward:localabstract:gnirehtet;tcp:31416
12-14 19:54:55.759  6561  6561 I ADB_SERVICES: service_to_fd shell:am broadcast -a com.genymob

**Comment by rom1v** [comment]
>    12-14 19:49:29.554 1160 1222 W BroadcastQueue: Permission Denial: broadcasting Intent { act=com.genymobile.gnirehtet.START flg=0x10 cmp=com.genymobile.gnirehtet/.GnirehtetControlReceiver } from null (pid=6196, uid=2000) requires android.permission.WRITE_SECURE_SETTINGS due to receiver com.genymobile.gnirehtet/.GnirehtetControlReceiver

See https://github.com/Genymobile/gnirehtet/issues/5.

**Comment by Agoriro** [comment]
Hello everyone, in my case I did not show the dialog of the vpn, I have a Huawei Mate 10 pro with Android 9 and the solution to show me the dialog was to go to settings -> Battery -> Applications start. Search for gnirehtet and place it manually

**Comment by Orthetrum** [comment]
> Hello everyone, in my case I did not show the dialog of the vpn, I have a Huawei Mate 10 pro with Android 9 and the solution to show me the dialog was to go to settings -> Battery -> Applications start. Search for gnirehtet and place it manually

I had to create a GitHub account just to thank you personally.

**Comment by Armanta** [comment]
> 
> 
> Hello everyone, in my case I did not show the dialog of the vpn, I have a Huawei Mate 10 pro with Android 9 and the solution to show me the dialog was to go to settings -> Battery -> Applications start. Search for gnirehtet and place it manually

Same things on Huawei P20.Big thanks

**Comment by breekoy** [comment]
> Hello everyone, in my case I did not show the dialog of the vpn, I have a Huawei Mate 10 pro with Android 9 and the solution to show me the dialog was to go to settings -> Battery -> Applications start. Search for gnirehtet and place it manually

Works on my Huawei Nova 3i too. Thank you!

**Comment by rom1v** [comment]
Should be fixed by f8681809a28a0073f791f0d9eae65da02ae73c3d.
